src/compiler/zx_calculus.jl

```diff
@@ -1,14 +1,72 @@
-using ZXCalculus
+using ZXCalculus, LightGraphs
+using ZXCalculus: qubit_loc
 
 function optimize!(ir::YaoIR)
     circ = to_ZX_diagram(ir)
-    clifford_simplify!(circ)
-    ir = to_YaoIR(circ)
+    circ = clifford_simplify(circ)
+    ir.body = to_YaoIR(circ)
     return ir
 end
 
+function to_YaoIR(circ::ZXDiagram{T, P}) where {T, P}
```

Comment:
you could just define a constructor of `YaoIR` here.

Reply:
This function only generates an IR not a YaoIR.  It should be renamed.

Reply:
but you reused some code from YaoIR constructor. It'd be better to make this either a constructor of `YaoIR` or a constructor for `IR`. `to_<node>` convention is used specifically for the parser side in YaoLang.